### ISSUE TITLE
Sonarr not adding to queue torrents deleted and re-added

### ISSUE THREAD

**Issue by rubasace** [Sonarr/Sonarr#3765]
**Describe the bug**
Sonarr doesn't add to queue torrents that have previously been downloaded and deleted. 

I'm downloading an episode that comes in a `.rar` file. After completion, RTorrent will extract it automatically into a subfolder (completion event script that uses unrarall script). 

When downloading the first time Sonarr is able to track the download in the `Queue` section and after the file gets properly extracted, it imports it automatically (all good!!).

The problem arises if then I delete the episode and the download data and search again for the episode (both automatically and manually), the download doesn't appear again in the Queue view if the torrent has previously been downloaded, and the episode never gets imported after extraction (but still can import it manually or even using `Quick Import`).

This isn't a big deal, as there aren't many scenarios where you want to download the same release more than once, but I was running some tests for my scripts and found it out. It seems like Sonarr checks if the release has already been imported (using history??) and then it just doesn't track it anymore. Maybe it should take into account when releases get deleted?

**System Information**

 - Sonarr Version: 3.0.3.803
 - Operating System: Docker (linuxserver/sonarr)
 -  Mono version: 5.20.1.34

**Additional context**
It can be reproduced downloading any episode that comes compressed and extracting it after it finishes (torrent client plugin, manually, any script, it doesn't matter): First download, the torrent is tracked in the Queue view and after extraction it will get picked up. After second time downloaded, it won't be tracked and won't get added after extraction.

**Comment by markus101** [comment]
If the release is grabbed after it's been imported then it should appear in the queue again, if you're manually re-adding the same torrent to the client then Sonarr will treat it as imported. Deleting the episode file doesn't delete anything from history, if it did removing a file that was still in the client would just import it again.

Please upload the trace logs of this behaviour and link them here, they are definitely required when filing a bug.

**Comment by rubasace** [comment]
The issue is very easily reproducible:

1. Ask Sonarr to download one episode automatically (it goes into queue)
2. Wait for it to finish and import
3. Delete the imported file from Sonarr interface and the download from the torrent client (I'm using rTorrent)
4. Ask Sonarr to download the episode automatically (it should pick the same exact release again, this time doesn't appear in the queue).

To reproduce it I downloaded the file Rick.and.Morty.S02E03.Auto.Erotic.Assimilation.REPACK.1080p.WEBRip.AAC2.0.H.264-phr0sty.mkv, just clicking on the automatic download for episode 3 of second season. It downloaded and imported correctly, I deleted from Sonarr interface and deleted with data from RTorrent, clicked again on automatic download and same file was downloaded. Second time Queue doesn't show the file and after completion it does not get imported automatically.

Here is the link to the logs: https://drive.google.com/open?id=1gK4V5i2lWB_SpMPE5wiMjLQC_eUlX68t

**Comment by Ae3NerdGod** [comment]
For me, this definitely already happens. as in deleted releases DO get readded in the queue, the same release too. unless specifically blacklisted 


to retierate this is what i was doing:
1 delete release from client 
2 <mess with permissions and mounts to try and fix them>
3 either 
A) click the magnifying glass to re-search for missing episode or
B) delete the show and add it again in the new mount location
in both instances the queue section was working properly and imported properly (when i had good permissions)

 i know because ive changed my config to three different operating systems in the past couple of months and this behavior (the opposite behavior the op is describing) helped me sort out some permission issues with my download client.

PS i settled with (so far)
transmission-daemon, ubuntu 20 

tl;dr
works fine for me, maybe check your permissions or see if another download client works properly in order  to narrow down the issue?

**Comment by markus101** [comment]
> delete the show and add it again in the new mount location

It’s definitely working for you because by deleting the series you’re deleting all the history for that series as well, whether it works without deleting is the issue.

**Comment by rubasace** [comment]
@Ae3NerdGod it's not a permission issue, everything runs under the same user and group and only when re-downloading it happens, so it definitely has something to do with the way Sonarr and Radarr deal with torrent/RTorrent downloads (same happens in Radarr).

**Comment by markus101** [comment]
Tracking this issue under: https://github.com/Sonarr/Sonarr/issues/2393
